[IMP] MRP: edited/updated Semi-Finished products doc v15 (#2787)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/management/sub_assemblies.rst

Manage semi-finished products
=============================

A *semi-finished product*, also known as a *subassembly*, is a manufactured product that is used as
a component in another product’s bill of materials (BoM). Semi-finished products are used to
simplify complex :abbr:`BoMs (Bills of Materials)` or to more accurately represent a manufacturing
flow. A :abbr:`BoM (Bill of Materials)` that contains semi-finished products is referred to as a
*multilevel BoM*, where the main *top-level product* and its subassemblies are distinguished.

Configure semi-finished products
================================

To set up a multilevel :abbr:`BoM (Bill of Materials)`, the top-level product and semi-finished
products must be configured. Therefore, the first step is to create the semi-finished products and
their :abbr:`BoMs (Bills of Materials)`. Please refer to :doc:`this page <bill_configuration>` for
more information on how to set up :abbr:`BoMs (Bills of Materials)`.

.. image:: sub_assemblies/semifinished_product_bom.png
   :align: center

>    :alt: Bill of materials for a semi-finished product.

Reviewer: ```suggestion
   :alt: A bill of materials for a semi-finished product.
```

Create the top-level Bill of Materials (BoM)
============================================

After the semi-finished products are fully configured, navigate to :menuselection:`Manufacturing

> --> Products --> Products`, then :guilabel:`Create` the top-level product. Configure the product’s
> specifications as desired, and be sure to :guilabel:`Save`.

Reviewer: ```suggestion
--> Products --> Products`. Then, :guilabel:`Create` the top-level product. Configure the product’s
specifications as desired, and be sure to :guilabel:`Save`.
```

> Once the top-level product is configured, go to :guilabel:`Bills of Materials` and
> :guilabel:`Create` a :abbr:`BoM (Bill of Materials)` for the top-level product. Then, simply add
> the semi-finished products to this :abbr:`BoM (Bill of Materials)`, along with any other necessary
> components.

Reviewer: ```suggestion
Once the top-level product is configured, click the :guilabel:`Bill of Materials` smart button on
the product form and click :guilabel:`Create` to make a :abbr:`BoM (Bill of Materials)` for the
top-level product. Then, simply add the semi-finished products to this :abbr:`BoM (Bill of
Materials)`, along with any other necessary components.
```

.. image:: sub_assemblies/custom_computer_bom.png
   :align: center

>    :alt: Bill of materials for a top-level product, containing a subassembly.

Reviewer: ```suggestion
   :alt: Bill of materials for a top-level product, containing a subassembly component.
```

>    :alt: Bill of materials for a top-level product, containing a subassembly component.

Reviewer: ```suggestion
   :alt: A bill of materials for a top-level product that contains a subassembly component.
```

Manage production planning
==========================

There are several methods to manage manufacturing order automation for products with multilevel
:abbr:`BoMs (Bills of Materials)`.

.. note:: 
    Semi-finished products are specifically used to manage manufacturable products with multilevel
    BoMs. If a BoM is being created simply to organize components or bundle sellable products,
    using :doc:`Kits <kit_shipping>` is the more appropriate option.

To automatically trigger manufacturing orders for semi-finished products after confirming a
manufacturing order for the main product, there are two options:

> **Option 1 (recommended):** Create *Reordering Rules* for the semi-finished products and set both
> the minimum and maximum desired stock quantities to 0. To review how to set up Reordering Rules,
> refer to :doc:`this page <../../../inventory_and_mrp/purchase/products/reordering>`.

Reviewer: ```suggestion
**Option 1 (recommended):** Create *Reordering Rules* for the semi-finished products and set both
the minimum and maximum desired stock quantities to `0`. To review how to set up reordering rules,
refer to :doc:`this page <../../../inventory_and_mrp/purchase/products/reordering>`.
```

> **Option 1 (recommended):** Create *Reordering Rules* for the semi-finished products and set both
> the minimum and maximum desired stock quantities to '0'. To review how to set up reordering rules,
> refer to :doc:`this page <../../../inventory_and_mrp/purchase/products/reordering>`.
> 
> **Option 2:** Activate the *Replenish on Order (MTO)* and *Manufacture* routes under the
> :guilabel:`Inventory` tab of the semi-finished product’s product form.

Reviewer 2: Number of formatting suggestions here:
- Would first, format this as a bulleted list
- fix the apostrophe
- change '0' to `0` since it's input
- add `:guilabel:`'s to **Option 2** since these are checkboxes on the UI and they are being directly addressed to activate.

As well the `To review how to...` sentence might serve better as a `.. :seealso:` here, just like the one in the previous comment. Another option is to place an accumulated list of related documentation articles either at the beginning of the doc so the reader gets a heads up to be comfortable with those topics or at least have them open in a new tab. Up to you 🙃 

```suggestion
- **Option 1 (recommended):** Create *Reordering Rules* for the semi-finished products and set both
  the minimum and maximum desired stock quantities to `0`.

  .. seealso::
     - :doc:`Don't run out of stock with reordering rules
       <../../../inventory_and_mrp/purchase/products/reordering>`

- **Option 2:** Activate the :guilabel:`Replenish on Order (MTO)` and :guilabel:`Manufacture`
  routes under the :guilabel:`Inventory` tab of the semi-finished product's product form.
```

> Option 1 is more flexible than Option 2 and is therefore recommended. Reordering Rules do not

Reviewer: ```suggestion
Option 1 is more flexible than Option 2 and is therefore recommended. Reordering rules do not
```

directly link demand to replenishment, and therefore allow stocks to be unreserved and redirected

> to other orders, if necessary. The Replenish on Order (MTO) route creates a unique link between the

Reviewer 2: Suggestion since we're _referencing_ something on the UI but not in an instructional way. 
Feel free to ignore if you disagree.

```suggestion
to other orders, if necessary. The *Replenish on Order (MTO)* route creates a unique link between the
```

semi-finished and top-level products, exclusively reserving quantities for the confirmed top-level
manufacturing order.

> Regardless of the method chosen, semi-finished products must be fully manufactured before work can
> begin on the top-level product.

Reviewer: ```suggestion
Regardless of the method chosen, semi-finished products must be fully manufactured before
manufacturing can begin on the top-level product.
```

> .. image:: sub_assemblies/semifinished_on_MO.png

Reviewer 2: Replace underscores with hyphens.
As well, [media file names can not have capital letters](https://www.odoo.com/documentation/15.0/contributing/documentation/content_guidelines.html#media-files).
```suggestion
.. image:: sub_assemblies/semifinished-on-mo.png
```

   :align: center

>    :alt: Manufacturing order for a top-level product.

Reviewer: ```suggestion
   :alt: A manufacturing order for a top-level product.
```

Manage semi-finished products
=============================

A *semi-finished product*, also known as a *subassembly*, is a manufactured product that is used as
a component in another product's bill of materials (BoM). Semi-finished products are used to
simplify complex :abbr:`BoMs (Bills of Materials)` or to more accurately represent a manufacturing
flow. A :abbr:`BoM (Bill of Materials)` that contains semi-finished products is referred to as a
*multilevel BoM*, where the main *top-level product* and its subassemblies are distinguished.

Configure semi-finished products
================================

To set up a multilevel :abbr:`BoM (Bill of Materials)`, the top-level product and semi-finished
products must be configured. Therefore, the first step is to create the semi-finished products and
their :abbr:`BoMs (Bills of Materials)`.

.. seealso::

>    - :doc:`bill_configuration`

Reviewer 3: ```suggestion
   :doc:`bill_configuration`
```

the product form, then click :guilabel:`Create` to make a :abbr:`BoM (Bill of Materials)` for the
top-level product. Then, simply add the semi-finished products to this :abbr:`BoM (Bill of
Materials)`, along with any other necessary components.

.. image:: sub_assemblies/custom-computer-bom.png
   :align: center
   :alt: A bill of materials for a top-level product, containing a subassembly component.

Manage production planning
==========================

There are several methods to manage manufacturing order automation for products with multilevel
:abbr:`BoMs (Bills of Materials)`.

.. note::
    Semi-finished products are specifically used to manage manufacturable products with multilevel
    BoMs. If a BoM is being created simply to organize components or bundle sellable products,
    using :doc:`Kits <kit_shipping>` is the more appropriate option.

To automatically trigger manufacturing orders for semi-finished products after confirming a
manufacturing order for the main product, there are two options:

- **Option 1 (recommended):** Create *Reordering Rules* for the semi-finished products and set both
  the minimum and maximum desired stock quantities to `0`.

>   - .. seealso::

Reviewer 3: ```suggestion
  .. seealso::
```

>      - :doc:`../../purchase/products/reordering`

Reviewer 3: No need to put a single seealso ref in a bulleted list.
```suggestion
     :doc:`../../purchase/products/reordering`
```
